src/chkstat.c

```diff
@@ -570,14 +652,15 @@ safe_open(char *path, struct stat *stb, uid_t target_uid, bool *traversed_insecu
 fail_insecure_path:
 
   {
-    char linkpath[PATH_MAX];
-    char procpath[100];
-    snprintf(procpath, sizeof(procpath), "/proc/self/fd/%d", pathfd);
-    ssize_t l = readlink(procpath, linkpath, sizeof(linkpath) - 1);
-    if (l > 0)
-        linkpath[(size_t)l < sizeof(linkpath) ? (size_t)l : sizeof(linkpath) - 1] = '\0';
-    else
-        memcpy(linkpath, "ancestor", 9);
+    char linkpath[PATH_MAX] = "ancestor";
+    char procpath[PROC_PATH_SIZE];
+    int res = make_proc_path(pathfd, procpath);
+    if (res == 0)
+      {
+        ssize_t l = readlink(procpath, linkpath, sizeof(linkpath) - 1);
+        if (l > 0)
+          linkpath[(size_t)l < sizeof(linkpath) ? (size_t)l : sizeof(linkpath) - 1] = '\0';
```

Comment:
how about `min( (size_t)l, sizeof(linkpath) -1)` for better readability?

Reply:
Done